release/release/common.clj

```diff
@@ -0,0 +1,275 @@
+(ns release.common
+  (:require [cheshire.core :as json]
+            [clj-http.client :as http]
+            [clojure.string :as str]
+            [colorize.core :as colorize]
+            [environ.core :as env])
+  (:import [java.io BufferedReader File InputStreamReader]
+           org.apache.commons.io.FileUtils))
+
+(assert (str/ends-with? (env/env :user-dir) "/release")
+        "Please run release.clj from the `release` directory e.g. `cd release; clojure -m release`")
+
+(defn env-or-throw
+  "Fetch an env var value or throw an Exception if it is unset."
+  [k]
+  (or (get env/env k)
+      (throw (Exception. (format "%s is unset. Please set it and try again." (str/upper-case (str/replace (name k) #"-" "_")))))))
+
+(def ^String root-directory
+  "e.g. /Users/cam/metabase"
+  (.getParent (File. (env/env :user-dir))))
```

Comment:
`$USER_DIR` doesn't exist on Linux. Is this the same as `$PWD`?

Reply:
Environ merges in Java system properties as well as environment variables into the `env` map, so in this case `user-dir` comes from the JVM property `user.dir`